lib/iris/fileformats/cf.py

```diff
@@ -1043,17 +1043,25 @@ class CFReader:
     # TODO: remove once iris.experimental.ugrid.CFUGridReader is folded in.
     CFGroup = CFGroup
 
-    def __init__(self, filename, warn=False, monotonic=False):
-        self._dataset = None
-        self._filename = os.path.expanduser(filename)
+    def __init__(self, file_source, warn=False, monotonic=False):
+        # Ensure safe operation for destructor, should init fail.
+        self._own_file = False
+        if isinstance(file_source, str):
```

Comment:
I think this needs updating to take account of a *Path* argument,
since https://github.com/SciTools/iris/pull/3917 ??

Reply:
Loader sources are converted from Path to str by `iris.io`, [here](https://github.com/SciTools/iris/blob/0d0116f9de420530a10fe359a267fb74e8187d17/lib/iris/io/__init__.py#L128-L129).

There might still be some unfinished business about support for Path internally (cf #4523), but it won't affect this work.